scripts/powershell/sub-cleanup/sub-cleanup.ps1

```diff
@@ -0,0 +1,70 @@
+﻿# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Requires PowerShell modules Az and RemoteSigned excecution policy
+#
+# Commands to make this happen:
+#   Install-Module -Name Az -AllowClobber
+#   Set-ExecutionPolicy RemoteSigned
+
+function Print-Help 
+{
+    Write-Host @"
+====================== sub-cleanup.ps1 ======================
+Cleans up non-compliant resource groups in a specified Azure
+subscription.  Complient resource groups are named with a
+valid Microsoft alias or using the format: alias-<identifier>
+
+Usage:
+    sub-cleanup.ps1 <subscription Id> <exception file>
+
+Notes:
+    The optional exception file is a text file listing
+    resource groups to be skipped in the cleanup.
+=============================================================
+"@
+}
+
+if ($args.Count -lt 1)
+{
+    Print-Help
+}
+
+# Load exceptions file if one was provided
+$exceptions = @()
+if ($args.Count -eq 2)
+{
+    Write-Host "Using exception file $($args[1])"
+    $exceptions = Get-Content $args[1]
+}
+
+# Connect and select active subscription
+$sub = $args[0]
+Connect-AzAccount
+Select-AzSubscription -Subscription $sub
+
+# Loop through the resource groups and delete any with non-compliant names
+$resourceGroups = Get-AzResourceGroup | Sort ResourceGroupName
+foreach ($resourceGroup in $resourceGroups)
+{
+    # skip exceptions
+    if ($exceptions -contains $resourceGroup.ResourceGroupName)
+    {
+        Write-Host " Skipping exception resource group: $($resourceGroup.ResourceGroupName)"
+        continue
+    }
+
+    # check compliance (formatting then valid alias) and skip if compliant
+    if ($resourceGroup.ResourceGroupName -match "^((t-|a-|v-)?[a-z,A-Z]{3,15})(-{1}.*)?$" -and (Get-AzAdUser -UserPrincipalName "$($Matches.1)@microsoft.com"))
```

Comment:
It might make sense to cache the outputs of `Get-AzAdUser` in the event that the same user has multiple resource groups that match this criteria (e.g. `djurek-testresources-storage`, `djurek-testresources-keyvault`, `djurek-testresources-etc`) you'd avoid having to check the same alias multiple times. 

It's also possible that aliases can be associated with different email domains (e.g. ntdev.microsoft.com)... I'm aware of one such case related to codeowners. Sadly the `Get-AzADUser`'s `-StartsWith` parameter seems to match on the display name of the user so we might have to make multiple calls to `Get-AzADUser` with different domain suffixes to know whether a prefix is not a valid alias. 

Hardcoding `@microsoft.com` limits the reusability of this script. It could be a script parameter and that would make it easier for others to use it. Given the potential for different account domains it might make sense to have this be a list sorted by the domain most likely to hold the alias. (e.g. `microsoft.com`, `ntdev.microsoft.com`, not sure what others are)